language/move-binary-format/src/file_format.rs

```diff
@@ -249,8 +249,25 @@ pub struct StructHandle {
     /// For any instantiation of this type, the abilities of this type are predicated on
     /// that ability being satisfied for all type parameters.
     pub abilities: AbilitySet,
-    /// The type formals (identified by their index into the vec) and their constraints
-    pub type_parameters: Vec<AbilitySet>,
+    /// The type formals (identified by their index into the vec)
+    pub type_parameters: Vec<StructTypeParameter>,
+}
+
+impl StructHandle {
+    pub fn type_param_constraints(&self) -> impl ExactSizeIterator<Item = AbilitySet> + '_ {
```

Comment:
Do you need the `'_` there? If so I find that rather amusing, since it feels like just a placeholder for inference

Reply:
Yeah, it's needed because the iterator captures data from `self` and by default there's an implicit `'static` bound for the `impl` so you need an explicit `'_` to relax that.

Reply:
nuts, did not know that